src/modules/auth/token.service.ts

```diff
@@ -64,4 +69,28 @@ export class TokenService {
   async invalidateRefreshToken(id: string) {
     return this.refreshTokenRepository.deleteRefreshTokenById(id);
   }
+
+  async handleOriginCheck(req: Request, res: Response, next: NextFunction) {
+    let token = null;
+    const requestOrigin = req.headers['origin'];
+
+    if (req.headers['authorization']) {
+      token = req.headers['authorization'].replace('Bearer ', '');
+    } else {
+      token = req.cookies.token
+    }
+
+    if (token) {
+      const decodedToken = this.jwtService.decode(token) as TokenPayload;
+      if (decodedToken.supportedOrigins.includes(requestOrigin) ||
```

Comment:
A continuation of the implementation on my first comment, I think this is better logic:
```suggestion
      if (decodedToken.origin === requestOrigin || requestOrigin === undefined)
```
The first case is for browsers which are setting the origin header and the second case is for servers which are not setting the origin header. Does this make sense?